python/ql/src/semmle/python/frameworks/internal/AutomatedModeling.qll

```diff
@@ -0,0 +1,1901 @@
+    or
+    spec = "Django::Views::View" and
+    qualifier = "openstack_dashboard.dashboards.admin.metadata_defs.views.ManageResourceTypes"
+    or
+    spec = "Django::Views::View" and qualifier = "horizon.forms.views.ModalFormView"
+    or
+    spec = "Django::Views::View" and
+    qualifier = "openstack_dashboard.dashboards.project.networks.views.IndexView"
+    or
+    spec = "Django::Views::View" and
+    qualifier = "openstack_dashboard.dashboards.project.networks.views.CreateView"
+    or
+    spec = "Django::Views::View" and
+    qualifier = "openstack_dashboard.dashboards.project.networks.views.UpdateView"
+    or
+    spec = "Django::Views::View" and
+    qualifier = "openstack_dashboard.dashboards.project.networks.views.DetailView"
+    or
+    spec = "Django::Forms::Field" and qualifier = "horizon.forms.fields.IPField"
+    or
+    spec = "Django::Forms::Field" and qualifier = "horizon.forms.fields.MultiIPField"
+    or
+    spec = "Django::Forms::Field" and qualifier = "horizon.forms.fields.MACAddressField"
+    or
+    spec = "Django::Forms::Field" and qualifier = "horizon.forms.fields.ThemableChoiceField"
+    or
+    spec = "Django::Forms::Field" and qualifier = "horizon.forms.fields.DynamicChoiceField"
+    or
+    spec = "Django::Forms::Field" and qualifier = "horizon.forms.fields.ThemableDynamicChoiceField"
+    or
+    spec = "Django::Forms::Field" and qualifier = "horizon.forms.fields.DynamicTypedChoiceField"
+    or
+    spec = "Django::Forms::Field" and
+    qualifier = "horizon.forms.fields.ThemableDynamicTypedChoiceField"
+    or
+    spec = "Django::Forms::Field" and qualifier = "horizon.forms.fields.ExternalFileField"
+    or
+    spec = "Django::Forms::Field" and qualifier = "horizon.forms.DynamicChoiceField"
+    or
+    spec = "Django::Forms::Field" and qualifier = "horizon.forms.DynamicTypedChoiceField"
+    or
+    spec = "Django::Forms::Field" and qualifier = "horizon.forms.ExternalFileField"
+    or
+    spec = "Django::Forms::Field" and qualifier = "horizon.forms.IPField"
+    or
+    spec = "Django::Forms::Field" and qualifier = "horizon.forms.MACAddressField"
+    or
+    spec = "Django::Forms::Field" and qualifier = "horizon.forms.MultiIPField"
+    or
+    spec = "Django::Forms::Field" and qualifier = "horizon.forms.ThemableChoiceField"
+    or
+    spec = "Django::Forms::Field" and qualifier = "horizon.forms.ThemableDynamicChoiceField"
+    or
+    spec = "Django::Forms::Field" and qualifier = "horizon.forms.ThemableDynamicTypedChoiceField"
+    or
+    spec = "Django::Forms::Field" and qualifier = "horizon.forms.Field"
+    or
+    spec = "Django::Forms::Field" and qualifier = "horizon.forms.FileField"
+    or
+    spec = "Django::Forms::Field" and qualifier = "horizon.forms.BooleanField"
+    or
+    spec = "Django::Forms::Field" and qualifier = "horizon.forms.IntegerField"
+    or
+    spec = "Django::Forms::Field" and qualifier = "horizon.forms.ChoiceField"
```

Comment:
Although less precise, would a wildcard like `qualifier.matches("horizon.forms.%Field")` make the code a bit shorter.

Reply:
I believe this part is essentially autogenerated, so making such changes would require refining the output. In that case, just accepting the length (and the higher precision) is probably best for now...

Reply:
Yep, this data is generated, and not hand written :smile: